bundlewrap/repo.py

```diff
@@ -541,88 +541,131 @@ def _build_node_metadata(self, blame=False):
             # until none of them return DONE anymore (indicating that they're
             # just waiting for another metaproc to maybe insert new data,
             # which isn't happening if none return DONE)
-            metaproc_returned_DONE = False
+
+
+            print('populate_dependencies')
+            
+            # collect all metaprocs
+            metaprocs = {}
             for node_name in list(self._node_metadata_partial):
+                node = self.get_node(node_name)
+                for name, func in node.metadata_processors:
+                    func.__setattr__('__node', node)
+                    metaprocs[name] = func
```

Comment:
You're mixing things up here, overwriting metadata processors with the same on different nodes. Get rid of the setattr and use a two-tuple like `blacklisted_metaprocs`.